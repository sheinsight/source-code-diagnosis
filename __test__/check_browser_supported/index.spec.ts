import { expect, test } from "vitest";
import { fileURLToPath } from "node:url";
import path, { dirname } from "node:path";
import { checkBrowserSupportedWithSourceCode, checkBrowserSupported } from "../../index.js";

const __dirname = dirname(fileURLToPath(import.meta.url));

test("There are 4 syntaxes under normal that are incompatible under chrome 40", async () => {
	const cwd = path.resolve(__dirname, "fixtures", "normal");

	const response = await checkBrowserSupported({ chrome: "40" }, { cwd });

	expect(response.length).toBe(4);
});

test("There are 3 syntaxes under normal that are incompatible under chrome 45", async () => {
	const cwd = path.resolve(__dirname, "fixtures", "normal");

	const response = await checkBrowserSupported({ chrome: "45" }, { cwd });

	expect(response.length).toBe(3);
});

test("Multiple occurrences of the same syntax results in multiple calculations.", async () => {
	const response1 = await checkBrowserSupportedWithSourceCode({ chrome: "70" }, "const b = cc ?? 3;", "test.ts");
	expect(response1.length).toBe(1);
	expect(response1).toMatchSnapshot();

	const response2 = await checkBrowserSupportedWithSourceCode(
		{ chrome: "70" },
		"const b = cc ?? 3; const c = cc ?? 3;",
		"test.ts",
	);
	expect(response2.length).toBe(2);
	expect(response2).toMatchSnapshot();
});
