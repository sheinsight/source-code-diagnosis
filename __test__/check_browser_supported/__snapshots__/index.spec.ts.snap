// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Multiple occurrences of the same syntax results in multiple calculations. 1`] = `
[
  {
    "astNode": {
      "loc": {
        "end": {
          "col": 18,
          "line": 1,
        },
        "start": {
          "col": 11,
          "line": 1,
        },
      },
      "span": {
        "end": 17,
        "start": 10,
      },
    },
    "compat": {
      "description": "空值合并运算符 (??)",
      "mdnUrl": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator",
      "name": "operators.nullish_coalescing",
      "support": {
        "chrome": "80",
        "chromeAndroid": "80",
        "deno": "1.0.0",
        "edge": "80",
        "firefox": "72",
        "firefoxAndroid": "72",
        "node": "14.0.0",
        "safari": "13.1",
        "safariIos": "13.1",
      },
      "tags": [
        "web-features:snapshot:ecmascript-2020",
      ],
    },
    "filePath": "test.ts",
    "name": "operators.nullish_coalescing",
  },
]
`;

exports[`Multiple occurrences of the same syntax results in multiple calculations. 2`] = `
[
  {
    "astNode": {
      "loc": {
        "end": {
          "col": 18,
          "line": 1,
        },
        "start": {
          "col": 11,
          "line": 1,
        },
      },
      "span": {
        "end": 17,
        "start": 10,
      },
    },
    "compat": {
      "description": "空值合并运算符 (??)",
      "mdnUrl": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator",
      "name": "operators.nullish_coalescing",
      "support": {
        "chrome": "80",
        "chromeAndroid": "80",
        "deno": "1.0.0",
        "edge": "80",
        "firefox": "72",
        "firefoxAndroid": "72",
        "node": "14.0.0",
        "safari": "13.1",
        "safariIos": "13.1",
      },
      "tags": [
        "web-features:snapshot:ecmascript-2020",
      ],
    },
    "filePath": "test.ts",
    "name": "operators.nullish_coalescing",
  },
  {
    "astNode": {
      "loc": {
        "end": {
          "col": 37,
          "line": 1,
        },
        "start": {
          "col": 30,
          "line": 1,
        },
      },
      "span": {
        "end": 36,
        "start": 29,
      },
    },
    "compat": {
      "description": "空值合并运算符 (??)",
      "mdnUrl": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator",
      "name": "operators.nullish_coalescing",
      "support": {
        "chrome": "80",
        "chromeAndroid": "80",
        "deno": "1.0.0",
        "edge": "80",
        "firefox": "72",
        "firefoxAndroid": "72",
        "node": "14.0.0",
        "safari": "13.1",
        "safariIos": "13.1",
      },
      "tags": [
        "web-features:snapshot:ecmascript-2020",
      ],
    },
    "filePath": "test.ts",
    "name": "operators.nullish_coalescing",
  },
]
`;
